unittests/testpublicassetattachments.py

```diff
@@ -0,0 +1,224 @@
+"""
+Test archivist
+"""
+
+from io import BytesIO
+from logging import getLogger
+from os import environ
+from unittest import TestCase, mock
+
+from archivist.archivistpublic import ArchivistPublic
+from archivist.constants import (
+    ROOT,
+    ASSETS_LABEL,
+    ASSETS_SUBPATH,
+    ATTACHMENTS_LABEL,
+    ASSETATTACHMENTS_LABEL,
+    ASSETATTACHMENTS_SUBPATH,
+)
+from archivist.logger import set_logger
+
+from .mock_response import MockResponse
+
+# pylint: disable=protected-access
+
+if "TEST_DEBUG" in environ and environ["TEST_DEBUG"]:
+    set_logger(environ["TEST_DEBUG"])
+
+LOGGER = getLogger(__name__)
+
+PROPS = {
+    "hash": {"alg": "SHA256", "value": "xxxxxxxxxxxxxxxxxxxxxxx"},
+    "mime_type": "image/jpeg",
+    "timestamp_accepted": "2019-11-07T15:31:49Z",
+    "size": 31424,
+}
+URL = "https://app.rkvst.io"
+ASSET_UUID = "b2678528-0136-4876-ad56-904e12c4b4c6"
+ASSET_ID = f"{URL}/{ROOT}/{ASSETS_SUBPATH}/public{ASSETS_LABEL}/{ASSET_UUID}"
+ATTACHMENT_UUID = "abcdef28-0136-4876-ad56-904e12c4b4c6"
+ATTACHMENT_ID = f"{ATTACHMENTS_LABEL}/{ATTACHMENT_UUID}"
+SUBPATH = (
+    f"{ASSETATTACHMENTS_SUBPATH}/{ASSETATTACHMENTS_LABEL}/"
+    f"public{ASSETS_LABEL}/{ASSET_UUID}/{ATTACHMENT_UUID}"
+)
+LOGGER.debug("Subpath %s", SUBPATH)
+RESPONSE = {
+    **PROPS,
+    "identity": ATTACHMENT_ID,
+}
+INFO = {
+    "hash": {
+        "alg": "SHA256",
+        "value": "e1105070ba828007508566e28a2b8d4c65d192e9eaf3b7868382b7cae747b397",
+    },
+    "identity": ATTACHMENT_ID,
+    "issuer": "",
+    "mime_type": "application/zip",
+    "size": 308,
+    "subject": "",
+    "tenantid": "",
+    "timestamp_accepted": "2022-03-01T09:25:43Z",
+    "scanned_status": "NOT_SCANNED",
+    "scanned_bad_reason": "",
+    "scanned_timestamp": "",
+}
+
+
+class TestPublicAssetAttachmentsBase(TestCase):
+    """
+    Test Archivist Attachments Create method
+    """
+
+    maxDiff = None
+
+    def setUp(self):
+        self.public = ArchivistPublic()
+
+    def tearDown(self):
+        self.public.close()
+
+    def test_assetattachments_str(self):
+        """
+        Test attachments str
+        """
+        self.assertEqual(
+            str(self.public.assetattachments),
+            "AssetAttachmentsClient()",
+            msg="Incorrect str",
+        )
+
+
+class TestPublicAssetAttachmentsDownload(TestPublicAssetAttachmentsBase):
+    """
+    Test Archivist Attachments Downloads method
+    """
+
+    maxDiff = None
+
+    def common_assetattachments_download(
+        self,
+        input_params,
+        expected_params,
```

Comment:
The behaviour of input_params vs expected-params is a bit opaque to me. Sure, it's not too hard to work it out given that this is a test but all the callers seem to pass in the same values for both, so isn't it redundant? In what circumstances would a caller pass in different values?

Reply:
this is called 3 times with different values for both so I am unclear about this?